app/async_database.py

```diff
@@ -0,0 +1,162 @@
+import contextlib
+import typing
+
+import aiosqlite
+import discord
+from database import UserProfile
+
+db_path = "demo.db"
+
+
+class AsyncDatabase:
+    """Class to store user profile and channel data asynchronously."""
+
+    def __init__(self, connection: aiosqlite.Connection) -> None:
+        self.connection = connection
+
+    async def enable_channel(self, channel: discord.TextChannel) -> None:
+        """Enable the game in a channel.
+
+        Arguments:
+        ---------
+        channel (discord.TextChannel): The channel that the game is to be enabled in
+
+        """
+        await self.connection.execute(
+            "INSERT INTO Guilds(id, channel) VALUES (?,?)",
+            (channel.guild.id, channel.id),
+        )
+        await self.connection.commit()
+
+    async def disable_channel(self, channel: discord.TextChannel) -> None:
+        """Disable the game in a channel.
+
+        Arguments:
+        ---------
+        channel (discord.TextChannel): The channel that the game is to be disabled in
+
+        """
+        await self.connection.execute("DELETE FROM Guilds WHERE id=? AND channel=?", (channel.guild.id, channel.id))
+        await self.connection.commit()
+
+    async def get_channels(self, guild: discord.Guild) -> list[discord.TextChannel]:
+        """Get all the channels that the game is enabled in for a guild.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild to check all enabled profiles in
+
+        """
+        async with (
+            self.connection.execute("SELECT channel FROM Guilds WHERE guild_id = ?", (guild.id)) as cursor,
+        ):
+            return [discord.Client.get_channel(row[0]) for row in cursor]
+
+    async def get_active_profiles(self, guild: discord.Guild) -> list[UserProfile]:
+        """Get all the profiles that the game is enabled in.
+
+        Arguments:
+        ---------
+        guild (discord.Guild): The guild to check all enabled profiles in
+
+        """
+        async with (
+            self.connection.execute("SELECT user_id, coins, cps FROM Users WHERE guild_id = ?", (guild.id)) as cursor,
+        ):
+            return [UserProfile(user=discord.Client.get_user(row[0]), coins=row[1], cps=row[2]) for row in cursor]
```

Comment:
Unfortunately `discord.Client.get_user` won't work here, you will need to take the client as an argument.

However, I don't really know what this could be used for... a leaderboard maybe? Potentially for removing all user profiles but we can do that anyways.